src/cpptools.ts

```diff
@@ -460,25 +485,32 @@ export class CppConfigurationProvider implements cpptools.CustomConfigurationPro
         }
 
         if (sysroot) {
-            flags.push('--sysroot=' + sysroot);
+            // Send sysroot with quoting for CppTools API V5 and below.
+            flags.push('--sysroot=' + shlex.quote(sysroot));
         }
 
         this.workspaceBrowseConfiguration = {
             browsePath: newBrowsePath,
-            standard,
             compilerPath: normalizedCompilerPath || undefined,
-            compilerArgs: flags || undefined
+            // Pass sysroot (without quote added) as the only compilerArgs for CppTools API V6 and above.
+            compilerArgs: !useFragments ? flags : sysroot ? [('--sysroot=' + sysroot)] : undefined,
```

Comment:
Can you do the v6 adding of flags without quotes in the block for sysroot above (line 487)?  I think it will be less confusing.